datasource/opennms-common/src/main/java/org/opennms/oce/datasource/common/inventory/script/AbstractScriptedInventory.java

```diff
@@ -0,0 +1,194 @@
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2019 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2019 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.oce.datasource.common.inventory.script;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+
+import javax.script.Invocable;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+import com.google.common.io.Resources;
+
+/**
+ * @author smith
+ *
+ */
+public abstract class AbstractScriptedInventory {
+
+    protected static final Logger LOG = LoggerFactory.getLogger(AbstractScriptedInventory.class);
+
+    private static final String DEFAULT_SCRIPT = "/inventory.groovy";
+
+    private static final String DEFAULT_SCRIPT_EXTENSION = "groovy";
+
+    private final boolean usingClasspathScript;
+
+    private final Invocable invocable;
+
+    private final String scriptPath;
+
+    private final ScriptEngineManager manager;
+
+    private final ScriptEngine engine;
+
+    private final long scriptCacheMillis;
+
+    private long configurationTimestamp;
+
+    private long scriptFileTimestamp;
+
+    public AbstractScriptedInventory(String scriptPath, long scriptCacheMillis, BundleContext bundleContext) {
+        if (scriptPath == null) {
+            throw new IllegalArgumentException("Null value for scriptFile.");
+        }
+
+        String script;
+        String scriptExtension;
+
+        this.scriptCacheMillis = scriptCacheMillis;
+
+        if (scriptPath.isEmpty()) {
+            // load default from classpath
+            usingClasspathScript = true;
+            if (bundleContext == null) {
+                throw new IllegalArgumentException("Path to Inventory Script must be provided.");
```

Comment:
This exception should probably be 'bundleContext cannot be null'.

Reply:
Actually, it's more useful to the user as written.
The reason is, bundleContext will be null (expected) if the code is not run inside the OSGi container - e.g. unit tests.
This block is entered when the path to the script is empty (which is permissible in the OSGi container - it is the default value).
So if they arrive here - they are running outside the container (a unit test) but have provided an empty value for the script path.

Reply:
Ok, makes sense.